### PR TITLE
Fixing golint warnings, batch 1

### PR DESCRIPTION
**Pull request by awly**
Mostly cosmetic changes:
- making receiver names consistent
- renaming `foo.FooBar` to `foo.Bar` (using package name as prefix)
- removing redundant `else` branches
- changing `a += 1` to `a++`

Updates https://github.com/gravitational/teleport/issues/3551

### CODE REVIEW

**Comment by webvictim** [comment]
This might sound weird but I don't 'like' the `AuthServer` -> `Server` and `LiteBackend` -> `Backend` changes. I feel like it makes the code less readable. I know that the idea is that you should be able to imply it from the package name, but I feel like being explicit helps a lot with `grep` and being able to read the code from a distance to see exactly what interface is intended to be used in a given situation. Maybe I've been here too long :)

The other changes (fix receiver names, incrementers, break loops early etc) absolutely make sense.

**Comment by webvictim** [comment]
That's totally fair, and honestly I use VS Code (which has `gopls` built in) so I can just hit `Go to Implementation` or `Go to References` and it'll find where it's defined anyway. I think I was more thinking for at-a-glance review on Github, which is harder without full code navigation and so forth. That comes with experience, though, and I'm not the one who's doing the majority of that review :)

I agree that consistency across everything is a more noble goal.

File: lib/auth/auth_with_roles.go [hunk 1]
```diff
@@ -36,33 +36,33 @@ import (
 	"github.com/tstranex/u2f"
 )
 
-// AuthWithRoles is a wrapper around auth service
+// WithRoles is a wrapper around auth service
 // methods that focuses on authorizing every request
-type AuthWithRoles struct {
-	authServer *AuthServer
+type WithRoles struct {
```
Comment (fspmarshall) [inline lib/auth/auth_with_roles.go:41]: Nit: `WithRoles` sounds like a function name to me.

**Review by fspmarshall — APPROVED** [review]

File: lib/auth/auth_with_roles.go (hunk 1 above)
Comment (awly) [reply lib/auth/auth_with_roles.go]: Renamed to `ServerWithRoles` :+1:

**Pull request merged**